miner/src/pool/scoring.rs

```diff
@@ -92,33 +92,69 @@ impl<P> txpool::Scoring<P> for NonceAndGasPrice where P: ScoredTransaction + txp
 	}
 
 	fn update_scores(&self, txs: &[txpool::Transaction<P>], scores: &mut [U256], change: scoring::Change) {
-		use self::scoring::Change;
-
-		match change {
-			Change::Culled(_) => {},
-			Change::RemovedAt(_) => {}
-			Change::InsertedAt(i) | Change::ReplacedAt(i) => {
-				assert!(i < txs.len());
-				assert!(i < scores.len());
-
-				scores[i] = *txs[i].transaction.gas_price();
-				let boost = match txs[i].priority() {
-					super::Priority::Local => 15,
-					super::Priority::Retracted => 10,
-					super::Priority::Regular => 0,
-				};
-				scores[i] = scores[i] << boost;
+
+		let boost = |i: usize| {
+			match txs[i].priority() {
+				super::Priority::Local => 15,
+				super::Priority::Retracted => 10,
+				super::Priority::Regular => 0,
+			}
+		};
+
+		let event = |scores: &mut [U256]| {
```

Comment:
maybe call this `penalize`, it took me few seconds to figure out what this function is about (explanation comment on what `event` means is below)